server/api/initial_data.py

```diff
@@ -821,6 +823,26 @@ def _delete_state_file():
         pass
 
 
+def _add_default_alert_templates(
+    db: server.api.db.sqldb.db.SQLDB, db_session: sqlalchemy.orm.Session
+):
+    for template in mlrun.template_config.pre_defined_templates:
+        record = db.get_alert_template(db_session, template.template_name)
+        if record is None or _alert_templates_differ(record, template):
+            db.store_alert_template(db_session, template)
+
+
+def _alert_templates_differ(template1, template2):
+    return (
+        template1.template_description != template2.template_description
+        or template1.description != template2.description
+        or template1.severity != template2.severity
+        or template1.trigger != template2.trigger
+        or template1.reset_policy != template2.reset_policy
+        or template1.criteria != template2.criteria
+    )
```

Comment:
Since you added `__eq__` to sub-classes of the template (probably so you can do this comparison), why not go all the way and put this code as `__eq__` in `AlertTemplate` and here just do `record != template`?

Reply:
because I didn't want to do extra comparisons that are not needed during mlrun-api startup